### PR TITLE
Add polygon selection option to SelectTool

### PR DESCRIPTION
**Pull request by saaaaaally**
- Add new option for polygon selection in SelectTool.
- Add keyin in SelectTool to allow testing of all different selection methods (box, line, pick, polygon)

polygon select key in demo

### CODE REVIEW

**Comment by bbastings** [comment]
Who is asking for this?

File: core/frontend/src/tools/SelectTool.ts [hunk 1]
```diff
@@ -729,4 +781,262 @@ export class SelectionTool extends PrimitiveTool {
...
+    let minHeight = 0;
+    let maxHeight = 0;
+    let first = true;
+
+    for (const pt of this._polygonPoints) {
+      const ecfPt: Point3d = (this.iModel && this.iModel.isGeoLocated) ? this.iModel.spatialToEcef(pt) : pt;
+      ecfPts.push(ecfPt.toJSONXYZ() as XYAndZ);
+      ecfXYZPts.push(ecfPt.toJSONXYZ() as XYAndZ);
+
+      if (this.iModel.isGeoLocated) {
```
Comment (bbastings) [inline core/frontend/src/tools/SelectTool.ts:925]: This is certainly not correct/needed? Why is this geo located specific code here?

**Comment by saaaaaally** [comment]
> Who is asking for this?

This is part of the inspection workflow in iTwin Experience, for improving ai detection crack selection. 
[Feature 1394017](https://dev.azure.com/bentleycs/SEG/_workitems/edit/1394017): Select tool in the 3D to be able to select in a polygon shape, not only rectangle

**Comment by bbastings** [comment]
> This is part of the inspection workflow in iTwin Experience, for improving ai detection crack selection. [Feature 1394017]
I suspect this won't be very robust for that use case. Because select works off reading the depth buffer and can only pick elements that light up a pixel if your "cracks" are composed of lot of tiny line strings it's not going to pick them all up...

**Comment by bbastings** [comment]
I really don't think we can add a new SelectionMethod. SelectTool is public and I know it's been sub-classed.

**Pull request closed without merging**